wordpress-mobile/WordPress-iOS
# Add prefetching to SiteMediaPageViewController and coalescing to ImageDownloader

- Add prefetching to `SiteMediaPageViewController`.
- Add coalescing to `ImageDownloader` that powers the prefetching. This change is also a prerequisite to removing `WPImageSource` because the improved downloader now has the same (and more) of its features.

To test:

- Open "Media"
- Tap on one of the items to open the item details
- Scroll left or right (test both direction)
- Verify that if you scroll slow enough, the images are prefetching in the direction you are scrolling

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/107fb4e9-4ede-4ea6-b15e-dee02a15a724

This is going to be tricky to test (maybe with a proxy?), but if, when you scroll to the side, the prefetching task is still pending, there should be no new request sent thanks to the coalescing support in `ImageDownloader.`

## Regression Notes
1. Potential unintended areas of impact: Site Media
2. What I did to test those areas of impact (or what existing automated tests I relied on): yes
3. What automated tests I added (or what prevented me from doing so): yes

PR submission checklist:


UI Changes testing checklist:
